src/lib.rs

```diff
@@ -1,5 +1,55 @@
 //! Winit is a cross-platform window creation and event loop management library.
 //!
+//! # Installation
+//!
+//! `winit` can be added to `Cargo.toml` as a dependency. It can be added via `cargo add`.
+//!
+//! ```bash
+//! $ cargo add winit
+//! ```
+//!
+//! To only enable the X11 backend on Free Unix[^unix] systems, disable default features
+//! and enable the `x11` feature.
+//!
+//! ```bash
+//! $ cargo add winit --no-default-features --features x11
+//! ```
+//!
+//! To only enable the Wayland backend on Free Unix systems, disable default features
+//! and enable the `wayland` feature.
+//!
+//! ```bash
+//! $ cargo add winit --no-default-features --features wayland
+//! ```
+//!
+//! These features have no effect on systems that are not Free Unix.
+//!
+//! ## Dependencies
+//!
+//! Dependencies on non-system libraries is managed through Cargo. For the X11
+//! backend, the following Ubuntu packages or their equivalents must be installed.
+//!
+//! - `libx11-dev`
+//! - `libxcb1-dev`
+//! - `libxi-dev`
+//! - `libxcbcommon-dev`
+//! - `libxcbcommon-x11-dev`
+//!
+//! For the Wayland backend, the following Ubuntu packages or their equivalents
+//! must be installed.
+//!
+//! - `libwayland-dev`
+//! - `libxcbcommon-dev`
+//!
+//! The "dev" packages are only needed for building binaries that use `winit`. On
+//! deployed system the non-`dev` equivalents need to be installed.
+//!
+//! The other backends (Windows, macOS, etc) do not have any dependencies on system libraries
+//! that don't already come with the operating system. However, note that the Windows backend
+//! only supports Windows 10 and above.
```

Comment:
Nit: Slight preference for putting this information in `winit::platform::*`

Reply:
And then linking to that if you really want to
```suggestion
//! ## Dependencies
//!
//! Dependencies on non-system libraries is managed through Cargo.
//! See the [`platform`] documentation for information on system dependencies.
```

Reply:
I'd like to avoid burying this information if we can. Everyone who uses winit should know what system libraries need to be installed.

Reply:
How about putting it in the top-level docs on `winit::platform`, then (along with the Tier 1 and Tier 2 target lists)?

Reply:
Idk., I just kinda think we're giving undue weight to Linux/other free unix needs here? Like, system libraries (and the OS version, since it's the same as `rustc`) aren't a problem on macOS/iOS, but there's a bunch of other very important concerns you need to be aware of there. Even more so with Android. But we still put this sort of information in `winit::platform`

Reply:
This is largely motivated by having to deal with persistent issues (read: NixOS) where people don't have packages set up properly on Linux. I'd like to have this information in as visible of a place as possible.